src/renderer/views/pool/PoolDetailsView.tsx

```diff
@@ -72,38 +77,61 @@ export const PoolDetailsView: React.FC = () => {
     }
   }, [oRouteAsset, setSelectedPoolAsset])
 
-  const routeAssetRD = RD.fromOption(oRouteAsset, () =>
-    Error(intl.formatMessage({ id: 'routes.invalid.asset' }, { asset }))
+  const routeAssetRD = useMemo(
+    () => RD.fromOption(oRouteAsset, () => Error(intl.formatMessage({ id: 'routes.invalid.asset' }, { asset }))),
+    [oRouteAsset, asset, intl]
   )
 
   const priceSymbol = useObservableState(selectedPricePoolAssetSymbol$, O.none)
 
   const priceRatio = useObservableState(priceRatio$, ONE_BN)
 
+  const isHistoryLoading = useObservableState(isHistoryLoading$, false)
+
   const poolDetailRD: PoolDetailRD = useObservableState(selectedPoolDetail$, RD.initial)
 
-  return FP.pipe(
-    RD.combine(routeAssetRD, poolDetailRD),
-    RD.fold(
-      renderInitialView,
-      renderPendingView,
-      (e: Error) => {
-        return <PoolStatus label={intl.formatMessage({ id: 'common.error' })} displayValue={e.message} />
-      },
-      ([asset, poolDetail]) => {
-        return (
-          <PoolDetails
-            asset={O.some(asset)}
-            depth={getDepth(poolDetail, priceRatio)}
-            volume24hr={get24hrVolume(poolDetail, priceRatio)}
-            allTimeVolume={getAllTimeVolume(poolDetail, priceRatio)}
-            totalSwaps={0 /* getTotalSwaps(history) */}
-            totalStakers={0 /* getTotalStakers(poolDetail) */}
-            priceUSD={getPriceUSD(poolDetail, priceRatio)}
-            priceSymbol={O.toUndefined(priceSymbol)}
-          />
+  const onRefreshData = useCallback(() => {
+    reloadSelectedPoolDetail()
+    reloadActionsHistory()
+  }, [reloadSelectedPoolDetail, reloadActionsHistory])
+
+  const refreshButtonDisabled = useMemo(() => {
+    return isHistoryLoading || FP.pipe(poolDetailRD, RD.isPending)
+  }, [isHistoryLoading, poolDetailRD])
+
+  const prev = useRef<React.ReactElement>()
+
+  return (
+    <>
+      <Styled.ControlsContainer>
+        <Styled.BackLink />
+        <RefreshButton clickHandler={onRefreshData} disabled={refreshButtonDisabled} />
+      </Styled.ControlsContainer>
+      {FP.pipe(
+        RD.combine(routeAssetRD, poolDetailRD),
+        RD.fold(
+          renderInitialView,
+          () => prev.current || renderPendingView(),
+          (e: Error) => {
+            return <PoolStatus label={intl.formatMessage({ id: 'common.error' })} displayValue={e.message} />
+          },
+          ([asset, poolDetail]) => {
+            return (prev.current = (
+              <PoolDetails
+                asset={O.some(asset)}
+                depth={getDepth(poolDetail, priceRatio)}
+                volume24hr={get24hrVolume(poolDetail, priceRatio)}
+                allTimeVolume={getAllTimeVolume(poolDetail, priceRatio)}
```

Comment:
Rethink about properties of `PoolDetail`. Many of them accepts `PoolDetail`, why not just passing `PoolDetail` down to `PoolDetails` and do all the calculation there. It will simplify its props and data handling `PoolDetailsView` ...

Reply:
yes. it makes sense. but here this values was not changed at all (just re-formatted) just because there are another PRs with changes for this area. i have touched that just to avoid conflicts with sibling PRs and would like to refactor this props-interface with another PR.